### DISCUSSION TITLE
What happened to nested data binding in Livewire 3?

### DISCUSSION THREAD

**Discussion by klasenma** [livewire/livewire#7900] [1. Help] [answered]
Hi, 

we're currently trying to migrate from Livewire 2 to 3. We used [nested binding](https://laravel-livewire.com/docs/2.x/data-binding#nested-binding) from time to time to keep track of filtered elements for example. With Livewire 3 this seems not to work anymore.. 

Here is a simplified example:
```php
<?php

namespace App\Livewire;

use Livewire\Component;
use Illuminate\View\View;
use Livewire\Attributes\Layout;
use Livewire\Attributes\Computed;

class NestedTestComponent extends Component
{
   public $state = [];

    public function getPaymentProviderProperty(): array
    {
        return [
            [
                'slug' => 'stripe',
                'name' => 'Stripe Payment',
            ],
            [
                'slug' => 'free',
                'name' => 'Free',
            ]
        ];
    }

    public function render(): View
    {
        return view('nested-test-component')
            ->layout('layouts.guest');
    }
}
```

```html
<div>
    @json($state)

    <br>
    <br>

    @foreach ($this->paymentProvider as $key => $paymentProvider)
        <x-form.inline-group label="{{ $paymentProvider['name'] }}" for="paymentProvider_{{ $key }}">
            <x-form.checkbox type="checkbox" wire:model.live="state.provider.{{ $key }}" value="{{ $paymentProvider['slug'] }}" />
        </x-form.inline-group>
    @endforeach
</div>
```

For Livewire 2 this gives the following result:
![livewire 2](https://github.com/livewire/livewire/assets/32737017/e8423cc9-259a-48f6-a86a-0091c001b6ee)

For Livewire 3 this:
![livewire 3](https://github.com/livewire/livewire/assets/32737017/688407a1-d6d5-4d23-83da-fba3b5429c2a)

We can't find something in the migration guide or even in the documentation. Seems like this feature just vanished?

Does anyone know about this? Will be time consuming to find all positions and change them (as some components are pretty complicated).

Thanks.

**Comment by danie-ramdhani** [comment, marked as answer]
here, i made a fix example for you:
https://wirebox.app/b/47rk0

**Comment by klasenma** [comment]
Thank you very much @danie-ramdhani , that works!

**Reply by danie-ramdhani** [reply]
mark as answerd when the discussion is done. thanks!
